console/src/components/Services/Events/CronTriggers/Container.tsx

```diff
@@ -39,8 +48,28 @@ const STContainer: React.FC<Props> = ({
 
   const currentTrigger = findScheduledTrigger(allTriggers, triggerName);
 
+  // TODO: This is a hack to deal with renaming cron triggers and stale props
+  // https://react-redux.js.org/api/hooks#stale-props-and-zombie-children
+  // Needs remodelling the state and careful handling of cron triggers rename
+  useEffect(() => {
+    if (currentTrigger) {
+      if (
+        typeof triggerPresence === 'object' &&
+        triggerPresence.type === 'timeout'
+      ) {
+        window.clearTimeout(triggerPresence.timeoutHandle);
+        setTriggerPresence('not-missing');
+      }
+    } else if (triggerPresence === 'not-missing') {
+      const timeoutHandle = window.setTimeout(() => {
+        setTriggerPresence('error-not-found');
+      }, 1200 /* arbitrary value */);
+      setTriggerPresence({ type: 'timeout', timeoutHandle });
+    }
+  }, [currentTrigger]);
```

Comment:
`setTriggerPresence` is a dependency here right?

Reply:
No, `setTriggerPresence` is stable.